docker/resource_docker_container.go

```diff
@@ -263,6 +263,32 @@ func resourceDockerContainer() *schema.Resource {
 				Set: resourceDockerHostsHash,
 			},
 
+			"ulimit": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"soft": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+						"hard": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+				Set: resourceDockerUlimitsHash,
```

Comment:
According to @catsby 's comment [here](https://github.com/terraform-providers/terraform-provider-docker/pull/21#discussion_r151534291) special hash functions are not needed anymore.
On #31 I'll remove the other ones as well.